server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -565,18 +565,10 @@ public final Searcher acquireSearcher(String source) throws EngineException {
         EXTERNAL, INTERNAL
     }
 
-    /**
-     * Returns the translog associated with this engine.
-     * Prefer to keep the translog package-private, so that an engine can control all accesses to the translog.
-     */
-    abstract Translog getTranslog();
-
     /**
      * Checks if the underlying storage sync is required.
      */
-    public boolean isTranslogSyncNeeded() {
-        return getTranslog().syncNeeded();
-    }
+    public abstract boolean isTranslogSyncNeeded();
```

Comment:
I think this all looks ok but I wonder if we should remove the `Translog` part from all these methods. They would read my cleaner to me and it would suddenly be an implementation detail for the engine if it's a translog.

Reply:
So that would mean:

- `isTranslogSyncNeeded` -> `isSyncNeeded`
- `syncTranslog` -> `sync`
- `ensureTranslogSynced` -> `ensureSynced`
- `acquireTranslogRetentionLock` -> `acquireRetentionLock`
- `newTranslogSnapshotFromMinSeqNo` -> `newSnapshotFromMinSeqNo`
- `estimateTranslogOperationsFromMinSeq` -> `estimateOperationsFromMinSeq`
- `getTranslogStats` -> `getStats`
- `getTranslogLastWriteLocation` -> `getLastWriteLocation`

So, I don't think this would be necessarily bad, but I do think it's not trivial, for example, we'd then have `sync` and `flush`, which to me is strange to have both (as an external person, is "sync" for synchronizing everything with Lucene?), we would also want `getStats` to return generic statistic information, not just from a translog? Does `getLastWriteLocation` return writes from Lucene or the Translog now?

I think maybe reorganizing `Engine` to be as small as possible is good, I think we might need better terminology than dropping "translog" from the names though, especially since we still return `Translog.Snapshot` and `Translog.Location` if we want to completely hide the translog. Right now even if we rename it's still not an implementation detail, because the classes are still Translog-specific.

For now I'll merge this and open a seperate issue to make Translog completely an implementation detail of the Engine.

Reply:
I think the `translog` is needed for now, the translog is not an implementation detail, it's a fundamental component of the system that other components go through the engine to interact with. A lot of these methods would not make sense if they didn't refer to the fact that they are doing something to the translog.

Reply:
+1 to what Lee and Jason said. It's a good idea but I don't think we're ready yet. @dakrone do you mind adding your renaming list to the https://github.com/elastic/elasticsearch/issues/31248 so it clarifies the issue?